core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -1081,18 +1090,38 @@ private void openAndSetInContainer(File file) {
       e.printStackTrace();
     }
     zimReaderContainer.setZimFile(file);
-    if (zimReaderContainer.getZimFileReader() != null) {
+    final ZimFileReader zimFileReader = zimReaderContainer.getZimFileReader();
+    if (zimFileReader != null) {
       if (mainMenu != null) {
-        mainMenu.onFileOpened(zimReaderContainer.getZimFileReader(), !urlIsInvalid());
+        mainMenu.onFileOpened(zimFileReader, !urlIsInvalid());
       }
       openMainPage();
-      presenter.loadCurrentZimBookmarksUrl();
+      safeDispose();
+      bookmarkingDisposable = Flowable.combineLatest(
+        newBookmarksDao.bookmarkUrlsForCurrentBook(zimFileReader),
+        webUrlsProcessor,
+        (bookmarkUrls, currentUrl) -> bookmarkUrls.contains(currentUrl)
+      ).subscribeOn(AndroidSchedulers.mainThread())
```

Comment:
```suggestion
      ).observeOn(AndroidSchedulers.mainThread())
```
I haven't touched Rx for some time. I may be wrong. :|

Reply:
No you are totally correct, I was coming down with a cold when writing this. Hmmm very strange that the I didn't crash while testing this. I would say the processor omissions force the downstream onto main because that is where it gets its items from